### DISCUSSION TITLE
Same page in different routes

### DISCUSSION THREAD

**Discussion by proko-sfl** [vercel/next.js#12465] [Help] [answered]
Hello! Is there a way to have the same page served in two (or more) different routes without having to duplicate its contents in both directories?
Something like :
- /[location]/properties/[property-slug]
- /properties/[property-id]

**Comment by mxck** [comment]
@proko-sfl hi!
You can use experimental feature `rewrites`. See this RFC with examples https://github.com/zeit/next.js/issues/9081

Next.js site already use this feature (check [link](https://github.com/zeit/next-site/blob/4fb515eb1b55dbd4cc369745106bb30f4b391725/next.config.js#L117)).

**Reply by proko-sfl** [reply]
Hi @mxck, this looks to be exactly what I need, thanks! 
I don't think we can use it right away since we should be going live in a couple of weeks but we can have some duplicate code until the RFC is ready for production. 
BTW do you know if I need canary to test experimental features like this?

**Reply by mxck** [reply]
You can test it right now :) Check next.js site link which I added above. They use `experimental` field in nextjs config.

**Reply by mxck** [reply]
@proko-sfl also be careful of `<Link>` usage. After rewrites you need to add `as` to all links (check rfc link above). 
So to your use case I thinks @StarpTech answer will be better :)

**Reply by proko-sfl** [reply]
Agreed!!

**Comment by StarpTech** [comment, marked as answer]
Just move your page in a separate file and reuse it in both places?

```jsx
import MyPage from 'pages/mypage'
default export MyPage
```

**Reply by proko-sfl** [reply]
Right! Thanks @StarpTech. Should be something like this: 
```
import MyPage from 'pages/mypage';

const MyOtherPage = ({property}) => <MyPage property={property} />

export async function getServerSideProps(ctx) {
 // get property using id from ctx
  return {
    props: {
      property
    },
  };
}

export default MyOtherPage;
```
